pallets/ajuna-tournament/src/traits.rs

```diff
@@ -12,11 +13,21 @@ pub trait EntityRank {
 	) -> sp_std::cmp::Ordering;
 }
 
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, Debug, Default, PartialEq)]
+pub enum TournamentPeriod {
+	#[default]
+	Inactive,
+	Active,
+	Claim,
+}
+
```

Comment:
When, I look at this, why don't we rewrite the `TournamentStruct` like this?

```rust
#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, Debug, Default, PartialEq)]
pub struct TournamentState<Balance> {
	pub id: TournamentId,
	pub reward: Balance,
	pub period: TournamentPeriod
}

pub enum TournamentPeriod {
	#[default]
	Inactive,
	Active,
	Claim,
}
```

It seems like a more intuitive to me than proxying the `TournamentState` enum to the `TournamentPeriod`.

Reply:
I am usually careful when different enum variants contain different fields, it is always a bit of a hassle to handle them. So I only do that when I can't avoid it.

Reply:
In the end I decided to just remove the TournamentPeriod on just use the TournamentState one, since adopting the struct solution meant complicating a bit more the logic in another parts of the code.